### PR TITLE
Cart

### PR DESCRIPTION
**Pull request by Thombuhl**

### CODE REVIEW

File: src/Cart.js [hunk 1]
```diff
@@ -1,21 +1,83 @@
...
-            { lineItem.product.name } { lineItem.quantity }
-          </li>
-            )
-      })
+
+class Cart extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
```
Comment (doobybooby) [inline src/Cart.js]: Do you need this.state?

File: db/User.js [hunk 2]
```diff
@@ -64,6 +64,7 @@ User.prototype.addToCart = async function({ product, quantity}){
 }
 
 User.prototype.getCart = async function(){
```
Comment (doobybooby) [inline db/User.js]: could you remove this console.log

File: src/SignIn.js [hunk 3]
```diff
@@ -10,23 +12,36 @@ class SignIn extends Component{
     super();
     this.state = {
       username: '',
-      password: ''
+      password: '',
+      loginAttempt: false,
+      error: false,
     };
     this.onChange = this.onChange.bind(this);
```
Comment (doobybooby) [inline src/SignIn.js:17]: I think it'll be a better naming convention if you do onChangeSomething

File: src/SignIn.js [hunk 4]
```diff
@@ -10,23 +12,36 @@ class SignIn extends Component{
...
     this.onSubmit = this.onSubmit.bind(this);
```
Comment (doobybooby) [inline src/SignIn.js:18]: same here i think we can do onFormSubmit to be more clear

File: src/store/cart.js [hunk 5]
```diff
@@ -1,25 +1,59 @@
 import axios from 'axios';
 
+const SET_CART = 'SET_CART';
+const UPDATE_LINEITEM_Q = 'UPDATE_CART';
+const DELETE_LINEITEM = 'DELETE_LINEITEM';
+
```
Comment (doobybooby) [inline src/store/cart.js:7]: I think we should separate the ACTIONS (above), from its handlers (below) and put them in different files) let me know what you think

**Review by doobybooby — COMMENTED** [review]
looks good to me

File: db/User.js (hunk 2 above)
Comment (Thombuhl) [reply db/User.js]: Console.log shall be removed

File: src/SignIn.js (hunk 4 above)
Comment (Thombuhl) [reply src/SignIn.js:18]: I will make changes, agreed.

File: src/store/cart.js (hunk 5 above)
Comment (Thombuhl) [reply src/store/cart.js:7]: Plan to do that after functioning is working.

File: src/SignIn.js (hunk 3 above)
Comment (Thombuhl) [reply src/SignIn.js:17]: I will update this

**Pull request merged**